tests/test_io.py

```diff
@@ -1,10 +1,26 @@
 import pkgutil
+import gzip
 
 from nose.tools import eq_
 
 from unittest import TestCase
 
-from edgePy.io import *  # Test import of __all__
+from edgePy.io import DataImporter, GroupImporter, get_dataset_path, get_open_function  # Test import of __all__
+
+
+class TestCommonFunctions(TestCase):
+
+    def test_get_open_function(self):
+
+        file_name = "blah.gz"
+        open_function, decode_required = get_open_function(filename=file_name)
+        eq_(gzip.open, open_function)
+        eq_(True, decode_required)
+
+        file_name = "blah.txt"
+        open_function, decode_required = get_open_function(filename=file_name)
+        eq_(open, open_function)
```

Comment:
A fragile comparison due to the test for equality on functions. Testing two functions are the same is actually a tricky problem in testing and best not done this way.

What you really want to test are that they behave as you intend!

To do this you could open a file encoded properly and ensure you can open and read contents (compare contents with `self.assertMultiLineEqual` in `unittest.TestCase`)